drivers/exec/driver_pre09.go

```diff
@@ -0,0 +1,36 @@
+package exec
+
+import (
+	"fmt"
+	"time"
+
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/nomad/drivers/shared/executor"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/plugins/drivers"
+)
+
+func (d *Driver) recoverPre0_9Task(config *drivers.TaskConfig, reattach *plugin.ReattachConfig) error {
+	config.ID = fmt.Sprintf("pre09-%s", uuid.Generate())
```

Comment:
Hm, what's this used for and why can it be arbitrary? Would be nice to have comments on the fields of this struct.

Reply:
So task ID is just an opaque identifier the driver uses to keep track of the task with. Its used in some debug logging. I thought having it prefixed might lend to help debugging in the future, but its probably premature and can totally just have a uuid.

The fields are documented pretty well in the proto. I was just lazy and never copied those docs to the go shim layer

Reply:
> I thought having it prefixed might lend to help debugging in the future, but its probably premature and can totally just have a uuid.

Nope! Prefix sounds nice! I was just curious if it mattered.

> The fields are documented pretty well in the proto.

Ah! Maybe a comment pointing in that direction then? I often jump-to-definition myself to the drivers structs and give up without thinking about the pbs.